src/use-case/delivery-cost.service.spec.ts

```diff
@@ -0,0 +1,348 @@
+      totalDiscountedAmount: 35,
+      totalDeliveryCostAfterDiscount: 665,
+    })
+  })
+
+  it('PKG 4: Testcase - getPackagePriceDiscount - found offer, but weight not within range', async () => {
+    //const deliveryCostService = new DeliveryCostService(
+    //inquiryService,
+    //offerRepository,
+    //cliTable,
+    //)
+
+    mockedOfferRepository.findByCode.mockResolvedValue(mockedOffer003)
+
+    const result = await deliveryCostService.getPackagePriceDiscount({
+      baseDeliveryCost: 100,
+      weightInKg: 4,
+      distanceInKm: 100,
+      offerCode: 'OFR003',
+      packageId: 'pkg4',
+    })
+
+    // Base Delivery Cost + (Package Total Weight * 10) +
+    // (Distance to Destination * 5) =
+    // 100 + (4 * 10) + (100 * 5) = 640
+    // applied offerCode: OFR003
+    // No discount, weight not within range
+
+    expect(result).toEqual({
+      packageId: 'pkg4',
+      totalDeliveryCostBeforeDiscount: 640,
+      totalDiscountedAmount: 0,
+      totalDeliveryCostAfterDiscount: 640,
+    })
+  })
+
+  it('PKG 5: Testcase - getPackagePriceDiscount - found offer, but distance not within range', async () => {
+    //const deliveryCostService = new DeliveryCostService(
+    //inquiryService,
+    //offerRepository,
+    //cliTable,
+    //)
+
+    mockedOfferRepository.findByCode.mockResolvedValue(mockedOffer003)
+
+    const result = await deliveryCostService.getPackagePriceDiscount({
+      baseDeliveryCost: 100,
+      weightInKg: 10,
+      distanceInKm: 4,
+      offerCode: 'OFR003',
+      packageId: 'pkg5',
+    })
+
+    // Base Delivery Cost + (Package Total Weight * 10) +
+    // (Distance to Destination * 5) =
+    // 100 + (10 * 10) + (4 * 5) = 220
+    // applied offerCode: OFR003
+    // No discount, distance not within range
+
+    expect(result).toEqual({
+      packageId: 'pkg5',
+      totalDeliveryCostBeforeDiscount: 220,
+      totalDiscountedAmount: 0,
+      totalDeliveryCostAfterDiscount: 220,
+    })
+  })
+
+  it('getTotalDeliveryCost - should get total delivery cost', async () => {
+    mockedInquireService.askBaseCostNoOfPackages.mockResolvedValue({
+      baseDeliveryCost: 100,
+      noOfPackages: 2,
+    })
+
+    mockedCliTable.initializeTable.mockReturnValue(
+      new Table({
+        head: ['Package Id', 'Discounted Amount', 'Total after discount'],
+        colWidths: [20, 20, 20],
+        wordWrap: true,
+      }),
+    )
+
+    mockedInquireService.askQuestionsForDeliveryCost
+      .mockResolvedValueOnce({
+        weightInKg: 5,
+        distanceInKm: 5,
+        offerCode: 'OFR001',
+        packageId: 'pkg1',
+      })
+      .mockResolvedValueOnce({
+        weightInKg: 10,
+        distanceInKm: 100,
+        offerCode: 'OFR003',
+        packageId: 'pkg3',
+      })
+
+    const result = deliveryCostService.getDeliveryCost()
+
+    expect(mockedInquireService.askBaseCostNoOfPackages).toHaveBeenCalledTimes(
+      1,
+    )
+
+    // expect(
```

Comment:
This already add in in [this PR](https://github.com/kenchoong/courrier-service/pull/5), better restructure the code for testing